training/metrics.py

```diff
@@ -27,83 +24,151 @@ def count_trainable_params(model):
   total_trainable_params = int(sum(np.prod(p.shape.as_list()) for p in object_identity.ObjectIdentitySet(weights)))
   return total_trainable_params
 
+def multiply_and_reduce(shape):
+  """Function calculates product between the shape (list) of the given layer and outputs a integer
+  eg: if shape is [10, 5, 3], the function is going to return ((10*5)*3).
 
-def _count_flops_conv2d(layer):
-  if layer.data_format == "channels_first":
-    input_channels = layer.input_shape[1]
-    output_channels, h, w, = layer.output_shape[1:]
-  elif layer.data_format == "channels_last":
-    input_channels = layer.input_shape[3]
-    h, w, output_channels = layer.output_shape[1:]
-  w_h, w_w = layer.kernel_size
+  Args:
+      shape (list): list of n values, typically 3
 
-  n_mult = h * w * output_channels * input_channels * w_h * w_w
-  n_add = w_h * w_w * input_channels * h * w * output_channels
+  Returns:
+      int 
+  """
+  return int(reduce(lambda x, y: x*y, shape))
```

Comment:
More generally, instead of shape you can use size. https://numpy.org/doc/stable/reference/generated/numpy.ndarray.size.html
Instead of `np.prod(a.shape[1:])` I would use `a[1:].size`

Reply:
Yes. np.prod works. However the a[1:].size not as the input is a tuple. Thanks